src/utilities/plan.ts

```diff
@@ -1,15 +1,72 @@
-import type { ActivityDirective } from '../types/activity';
+import type { ActivityDirective, ActivityDirectiveDB } from '../types/activity';
+import type { User } from '../types/app';
 import type { ArgumentsMap } from '../types/parameter';
-import type { DeprecatedPlanTransfer, Plan, PlanTransfer } from '../types/plan';
+import type { DeprecatedPlanTransfer, Plan, PlanSlim, PlanTransfer } from '../types/plan';
+import type { Simulation } from '../types/simulation';
+import effects from './effects';
 import { convertDoyToYmd } from './time';
 
-export function getPlanForTransfer(
-  plan: Plan,
-  activities: ActivityDirective[],
-  simulationArguments: ArgumentsMap,
-): PlanTransfer {
+export async function getPlanForTransfer(
+  plan: Plan | PlanSlim,
+  user: User | null,
+  progressCallback?: (progress: number) => void,
+  activities?: ActivityDirective[],
+  signal?: AbortSignal,
```

Comment:
Since this has a signal arg should it try/catch the abortable effect? Though maybe there isn't any sort of "cancel" yet exposed in the UI?

Reply:
The effect itself is already handling the abort error so there wouldn't necessarily need to be a try/catch here. Just a way to handle the case when the signal becomes aborted.